src/java/org/apache/cassandra/io/sstable/SSTableHeaderFix.java

```diff
@@ -315,7 +315,16 @@ void processFileOrDirectory(Path path)
     {
         Stream.of(path)
               .flatMap(SSTableHeaderFix::maybeExpandDirectory)
-              .filter(p -> Descriptor.fromFileWithComponent(new File(p)).right.type == SSTableFormat.Components.DATA.type)
+              .filter(p -> {
+                  try
+                  {
+                      return Descriptor.fromFileWithComponent(new File(p)).right.type == SSTableFormat.Components.DATA.type;
+                  }
+                  catch (IllegalArgumentException e)
+                  {
+                      return false;
```

Comment:
Maybe add a comment for future context that the intent is to ignore the `.keep` files?